src/components/Gauge.js

```diff
@@ -0,0 +1,263 @@
+      const percentOfTotal = dataPoint / this.props.chartTotal;
+      const endAngle = (percentOfTotal * 270) - 135;
+
+      const dataPointArc = d3.svg.arc()
+        .outerRadius(this.state.radius - this.props.activeOffset)
+        .innerRadius(this.state.radius - this.props.arcWidth)
+        .startAngle(endAngle * (Math.PI / 180))
+        .endAngle(endAngle * (Math.PI / 180));
+
+      return (
+        <circle
+          cx='0'
+          cy='0'
+          fill={this.props.dataPointColors[index]}
+          key={index}
+          r={this.props.dataPointRadius}
+          transform={'translate(' + dataPointArc.centroid() + ')'}
+        />
+      );
+    });
+  },
+
+  _renderDataLabel () {
+    const styles = this.styles();
+
+    if (this.props.showDataLabel) {
+      if (this.props.children) {
+        return (
+          <div
+            className='mx-gauge-data'
+            onClick={this._handleClick}
+            style={styles.center}
+          >
+            {this.props.children}
+          </div>
+        );
+      } else {
+        const color = this.props.colors[0];
+        const text = this.props.defaultLabelText;
+        const value = this.props.formatter(this.props.defaultLabelValue);
+
+        return (
+          <div
+            className='mx-gauge-data'
+            onClick={this._handleClick}
+            style={styles.center}
+          >
+            <div className='mx-gauge-data-value' style={[styles.value, { color }]}>
+              {value}
+            </div>
+            <div className='mx-gauge-data-label' style={styles.label}>
+              {text}
+            </div>
+          </div>
+        );
+      }
+    } else {
+      return null;
+    }
+  },
+
+  render () {
+    const position = 'translate(' + this.props.width / 2 + ',' + this.props.height / 2 + ')';
+    const fontSize = Math.min(this.props.width, this.props.height) * 0.2;
+    const styles = this.styles();
+
+    return (
+      <div
+        className='mx-gauge'
+        style={[styles.component, this.props.style, { fontSize, height: this.props.height, width: this.props.width }]}
+      >
+        {this._renderDataLabel()}
+        <svg
+          className='mx-gauge-svg'
+          height={this.props.height}
+          ref={this.props.id}
+          width={this.props.width}
+        >
+          <g className='mx-gauge-g' transform={position}>
+            {this._renderBaseArc()}
+            {this._renderArcs()}
+            {this._renderDataPoints()}
+          </g>
+        </svg>
+      </div>
+    );
+  },
+
+  styles () {
+    return {
+      component: {
+        position: 'relative',
+        fontFamily: StyleConstants.FontFamily
+      },
+      center: {
+        position: 'absolute',
+        top: '50%',
+        left: '50%',
+        textAlign: 'center',
+        transform: 'translate(-50%, -50%)'
+      },
+      label: {
+        color: StyleConstants.Colors.ASH,
+        fontSize: '0.4em',
```

Comment:
We're trying to use font sizes form the style constants to keep things consistent.  https://github.com/mxenabled/mx-react-components/blob/master/src/constants/Style.js  If you have questions on which size to use then you can get with Derek from product.